File: src/my_cpp_pkg/src/add_two_ints_server.cpp
#include "rclcpp/rclcpp.hpp"
#include "example_interfaces/srv/add_two_ints.hpp"

class AddTwoIntsServerNode : public rclcpp::Node
{
public:
    AddTwoIntsServerNode() : Node("add_two_ints_server")
    { 
        pServer = this->create_service<example_interfaces::srv::AddTwoInts>(
            "add_two_ints",
            std::bind(&AddTwoIntsServerNode::callbackAddTwoInts, this,
            std::placeholders::_1, std::placeholders::_2) 
        );

        RCLCPP_INFO(this->get_logger(), "Service server has been started");
    }

    ~AddTwoIntsServerNode()
    {
        //NOP
    }

private:
    std::shared_ptr<rclcpp::Service<example_interfaces::srv::AddTwoInts>> pServer;

    void callbackAddTwoInts(const std::shared_ptr<example_interfaces::srv::AddTwoInts::Request> request,
                            const std::shared_ptr<example_interfaces::srv::AddTwoInts::Response> response)
    {
        response->sum = request->a + request->b;
        RCLCPP_INFO(this->get_logger(), "%ld + %ld = %ld", request->a, request->b, response->sum);
    }
};

int main(int argc, char **argv)
{
    rclcpp::init(argc, argv);
    auto node = std::make_shared<AddTwoIntsServerNode>();
    rclcpp::spin(node);
    rclcpp::shutdown();
    return 0;
}

File: src/my_py_pkg/my_py_pkg/add_two_ints_server.py
import rclpy
from rclpy.node import Node
from example_interfaces.srv import AddTwoInts

class AddTwoIntsServerNode(Node):
    def __init__(self) -> None:
        super().__init__("add_two_ints_server")
        self.server = self.create_service(AddTwoInts, "add_two_ints",
                            self.callback_add_two_ints)
        self.get_logger().info("Server is started")
        
    def callback_add_two_ints(self, request, response):
        response.sum = request.a + request.b
        self.get_logger().info(f"{request.a} + {request.b} = {response.sum}")
        return response
        
def main(args = None):
    rclpy.init(args=args)
    node = Node("add_two_ints_server_node")
    node = AddTwoIntsServerNode()
    rclpy.spin(node)
    rclpy.shutdown()

if __name__ == "__main__":
    main()
